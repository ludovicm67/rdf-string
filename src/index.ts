export { sparql, SparqlTemplateResult, SparqlValue } from './lib/sparql'
