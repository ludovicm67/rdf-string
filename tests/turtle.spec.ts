import { blankNode, literal, namedNode } from '@rdfjs/data-model'
import { xsd } from '@tpluscode/rdf-ns-builders'
import { turtle } from '../src'

describe('turtle', () => {
  it('serializes named node', () => {
    // given
    const node = namedNode('http://example.com/')

    // when
    const str = turtle`${node}`.toString()

    // then
    expect(str).toEqual('<http://example.com/>')
  })

  it('serializes blank node', () => {
    // given
    const node = blankNode('foo')

    // when
    const str = turtle`${node}`.toString()

    // then
    expect(str).toEqual('_:foo')
  })

  it('serializes typed literal node', () => {
    // given
    const node = literal('foo', 'http://example.com/Datatype')

    // when
    const str = turtle`${node}`.toString()

    // then
    expect(str).toEqual('"foo"^^<http://example.com/Datatype>')
  })

  it('reduces known datatype URI to prefixed name', () => {
    // given
    const node = literal('foo', xsd.TOKEN)

    // when
    const str = turtle`${node}`.toString()

    // then
    expect(str).toEqual(`@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

"foo"^^xsd:TOKEN`)
  })

  it('serializes literal node with language', () => {
    // given
    const node = literal('foo', 'fr')

    // when
    const str = turtle`${node}`.toString()

    // then
    expect(str).toEqual('"foo"@fr')
  })
})
