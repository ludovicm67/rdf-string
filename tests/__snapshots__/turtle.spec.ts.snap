// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`turtle dataset interpolation combines multiple objects 1`] = `
"<http://example.com/S> <http://example.com/P> <http://example.com/O1> ,
      <http://example.com/O2> ,
      <http://example.com/O3> ."
`;

exports[`turtle dataset interpolation combines multiple objects for subsequent quads 1`] = `
"<http://example.com/S> <http://example.com/P> <http://example.com/O1> ,
      <http://example.com/O2> ,
      <http://example.com/O3> ."
`;

exports[`turtle dataset interpolation combines multiple predicates from same subsequent subject 1`] = `
"<http://example.com/S> <http://example.com/P1> <http://example.com/O1> ;
   <http://example.com/P2> <http://example.com/O2> ;
   <http://example.com/P3> <http://example.com/O3> ."
`;

exports[`turtle dataset interpolation correctly compresses output when prefixing names 1`] = `
"@prefix schema: <http://schema.org/> .

schema:S schema:P schema:O1 ,
      schema:O2 ,
      schema:O3 .
schema:S1 schema:P1 schema:O ;
   schema:P2 schema:O ."
`;

exports[`turtle dataset interpolation does not combine multiple objects for non-linear quads 1`] = `
"<http://example.com/S> <http://example.com/P> <http://example.com/O1> ;
   <http://example.com/P1> <http://example.com/O2> ;
   <http://example.com/P> <http://example.com/O3> ."
`;

exports[`turtle dataset interpolation does not combine multiple predicates for non-linear quads 1`] = `
"<http://example.com/S> <http://example.com/P> <http://example.com/O1> .
<http://example.com/S1> <http://example.com/P> <http://example.com/O2> .
<http://example.com/S> <http://example.com/P> <http://example.com/O3> ."
`;
